File: scripts/ask_ai.mjs
#!/usr/bin/env node
import fs from 'fs'
import path from 'path'

// Lire .env.local
let apiKey = process.env.NVIDIA_API_KEY || ''
if (!apiKey) {
  const envPath = path.resolve(process.cwd(), '.env.local')
  if (fs.existsSync(envPath)) {
    const lines = fs.readFileSync(envPath, 'utf8').split('\n')
    for (const line of lines) {
      const match = line.match(/^\s*NVIDIA_API_KEY\s*=\s*(.*?)\s*$/)
      if (match) {
        apiKey = match[1].replace(/["']/g, '')
        break
      }
    }
  }
}

if (!apiKey) {
  console.error('❌ Erreur: Variable NVIDIA_API_KEY introuvable dans .env.local')
  process.exit(1)
}

const args = process.argv.slice(2)
let prompt = ''
let filePath = ''

for (let i = 0; i < args.length; i++) {
  if (args[i] === '--file' || args[i] === '-f') {
    filePath = args[i + 1] || ''
    i++
  } else {
    prompt += (prompt ? ' ' : '') + args[i]
  }
}

let context = ''
if (filePath && fs.existsSync(filePath)) {
  context = fs.readFileSync(filePath, 'utf8')
}

if (!prompt && !context) {
  console.log('Utilisation: node scripts/ask_ai.mjs "Votre question de code..."')
  console.log('Option fichier: node scripts/ask_ai.mjs -f components/StockManager.tsx "Analyse ce code"')
  process.exit(0)
}

let fullContent = prompt || 'Analyse ce code et propose des améliorations.'
if (context) {
  fullContent = `--- CONTEXTE FICHIER (${filePath}) ---\n${context}\n\n--- QUESTION / INSTRUCTION ---\n${fullContent}`
}

console.log('🤖 [NVIDIA Nemotron 30B Reasoning] Analyse en cours...\n')

try {
  const response = await fetch('https://integrate.api.nvidia.com/v1/chat/completions', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${apiKey}`,
      Accept: 'application/json',
    },
    body: JSON.stringify({
      model: 'nvidia/nemotron-3-nano-omni-30b-a3b-reasoning',
      messages: [{ role: 'user', content: fullContent }],
      temperature: 0.6,
      top_p: 0.95,
      max_tokens: 65536,
      reasoning_budget: 16384,
      stream: false,
    }),
  })

  if (!response.ok) {
    const errorText = await response.text()
    console.error(`❌ Erreur API (${response.status}): ${errorText}`)
    process.exit(1)
  }

  const data = await response.json()
  const choice = data.choices?.[0] || {}
  const message = choice.message || {}

  if (message.reasoning_content) {
    console.log('🧠 --- RAISONNEMENT DU MODÈLE ---')
    console.log(message.reasoning_content)
    console.log('\n' + '='.repeat(60) + '\n')
  }

  console.log('💡 --- RÉPONSE / CODE DU MODÈLE ---')
  console.log(message.content || 'Aucune réponse.')
} catch (err) {
  console.error('❌ Exception:', err.message || err)
}

File: scripts/ask_ai.py
#!/usr/bin/env python3
"""
Script CLI pour interroger le modèle NVIDIA Nemotron Reasoning.
Utilisation:
  python scripts/ask_ai.py "Explique comment optimiser cette fonction..."
  python scripts/ask_ai.py --file components/StockManager.tsx "Analyse ce fichier"
"""

import os
import sys
import json
import argparse
import requests
from dotenv import load_dotenv

# Charger les variables d'environnement depuis .env.local ou .env
load_dotenv('.env.local')
load_dotenv('.env')

API_KEY = os.getenv('NVIDIA_API_KEY')
INVOKE_URL = "https://integrate.api.nvidia.com/v1/chat/completions"

def ask_nemotron(prompt: str, context: str = "", model: str = "nvidia/nemotron-3-nano-omni-30b-a3b-reasoning"):
    if not API_KEY:
        print("❌ Erreur: Variable NVIDIA_API_KEY introuvable.")
        print("👉 Veuillez ajouter NVIDIA_API_KEY=nvapi-... dans votre fichier .env.local")
        sys.exit(1)

    full_content = prompt
    if context:
        full_content = f"--- CONTEXTE CODE ---\n{context}\n\n--- QUESTION / INSTRUCTION ---\n{prompt}"

    headers = {
        "Authorization": f"Bearer {API_KEY}",
        "Accept": "application/json",
        "Content-Type": "application/json",
    }

    payload = {
        "model": model,
        "messages": [
            {
                "role": "user",
                "content": full_content
            }
        ],
        "temperature": 0.6,
        "top_p": 0.95,
        "max_tokens": 65536,
        "reasoning_budget": 16384,
        "stream": False
    }

    print(f"🤖 [NVIDIA Nemotron 30B Reasoning] Analyse en cours...")
    try:
        res = requests.post(INVOKE_URL, headers=headers, json=payload, timeout=120)
        if res.status_code != 200:
            print(f"❌ Erreur API ({res.status_code}): {res.text}")
            return

        data = res.json()
        choice = data.get("choices", [{}])[0]
        message = choice.get("message", {})
        content = message.get("content", "")
        reasoning = message.get("reasoning_content", "")

        if reasoning:
            print("\n🧠 --- RAISONNEMENT DU MODÈLE ---")
            print(reasoning)

        print("\n💡 --- RÉPONSE / CODE GÉNÉRÉ ---")
        print(content)

    except Exception as e:
        print(f"❌ Exception: {e}")

def main():
    parser = argparse.ArgumentParser(description="Assistant de code IA avec NVIDIA Nemotron Reasoning")
    parser.add_argument("prompt", nargs="?", default="", help="Votre question ou instruction de code")
    parser.add_argument("--file", "-f", help="Fichier de code à passer en contexte")
    args = parser.parse_args()

    context = ""
    if args.file and os.path.exists(args.file):
        with open(args.file, "r", encoding="utf-8", errors="ignore") as f:
            context = f.read()

    if not args.prompt and not context:
        print("Usage: python scripts/ask_ai.py \"Votre question de code...\"")
        print("Option fichier: python scripts/ask_ai.py --file components/StockManager.tsx \"Trouve les bugs\"")
        return

    ask_nemotron(args.prompt or "Analyse ce code et propose des améliorations.", context)

if __name__ == "__main__":
    main()
